pkg/collector/py/api.go

```diff
@@ -241,6 +243,14 @@ func isNone(o *C.PyObject) bool {
 	return int(C._is_none(o)) != 0
 }
 
+func stringRepresentation(o *C.PyObject) string {
+	repr := C._PyObject_Repr(o)
+	if repr != nil {
+		return C.GoString(C.PyString_AsString(repr))
+	}
+	return ""
```

Comment:
if `repr` is `nil`, the error flag is set on the python interpreter, so we should either:
* handle/clear the exception before the next cpython call, or
* return `nil` from the top-level go function (the one that's called from the C extension code), the C function should then return `NULL` so that the exception is raised when the C function returns.